tests/test_data_adapter.py

```diff
@@ -161,6 +215,41 @@ def test_geodataset(geoda, geodf, ts, tmpdir):
     # assert match in error
 
 
+def test_geodataset_unit_attrs(artifact_data: DataCatalog, tmpdir: str):
+    gtsm_dict = {"gtsmv3_eu_era5": artifact_data.sources["gtsmv3_eu_era5"].to_dict()}
+    attrs = {
+        "waterlevel": {
+            "long_name": "sea surface height above mean sea level",
+            "unit": "meters",
+        }
+    }
+    gtsm_dict["gtsmv3_eu_era5"].update(dict(attrs=attrs))
+    artifact_data.from_dict(gtsm_dict)
+    gtsm_geodataarray = artifact_data.get_geodataset("gtsmv3_eu_era5")
+    assert gtsm_geodataarray.attrs["long_name"] == attrs["waterlevel"]["long_name"]
+    assert gtsm_geodataarray.attrs["unit"] == attrs["waterlevel"]["unit"]
+
+
+def test_geodataset_unit_conversion(artifact_data: DataCatalog):
+    gtsm_geodataarray = artifact_data.get_geodataset("gtsmv3_eu_era5")
+    gtsm_dict = {"gtsmv3_eu_era5": artifact_data.sources["gtsmv3_eu_era5"].to_dict()}
+    gtsm_dict["gtsmv3_eu_era5"].update(dict(unit_mult=dict(waterlevel=1000)))
+    datacatalog = DataCatalog()
+    datacatalog.from_dict(gtsm_dict)
+    gtsm_geodataarray1000 = datacatalog.get_geodataset("gtsmv3_eu_era5")
+    assert gtsm_geodataarray1000.equals(gtsm_geodataarray * 1000)
```

Comment:
If we multiply like this, should the unit name name also be adjusted? (no is a perfectly acceptable answer here, I'm genuinely not sure)

Reply:
In theory yes, but we are only testing the multiplication of the units. Changing the unit name is already covered in other tests.